### DISCUSSION TITLE
Project Updates: Ongoing Work and Current Progress... sometimes

### DISCUSSION THREAD

**Discussion by werwolf2303** [SpotifyXP/SpotifyXP#61] [Announcements]
Here, you will receive updates on what is currently being worked on.

Thanks to [Jri-creator](https://github.com/Jri-creator) for reminding me about the discussions feature on GitHub!

**Comment by werwolf2303** [comment]
![shot](https://github.com/user-attachments/assets/68c3b845-9d83-4165-ad74-e57952357f57)
oops

**Comment by werwolf2303** [comment]
I'm currently working on adding oauth support so that zeroconf is no longer necessary.

https://github.com/librespot-org/librespot-java/commit/564430687554142752ff39cb04a0f827c8e12db2

**Comment by Jri-creator** [comment]
Any progress? By the way, that OAuth saves my bacon and wallet!

**Comment by werwolf2303** [comment]
**GitHub nuked the token that was used for the updater**. I'm working on a resolution

**Comment by Jri-creator** [comment]
I've got to say, I love the new Setup image, logo, and new Log Tools. Keep up the great work!

**Comment by werwolf2303** [comment]
![NTify Artist new stuff](https://github.com/user-attachments/assets/69f1b70e-a6be-4387-84ca-8d98c7bf51b1)

Currently working on adding "Discovered On" and "Related Artists" to the artist view. I added the followers and finally centered the artist image

**Comment by werwolf2303** [comment]
I'm working on a caching system right now (Don't worry for your HDD systems. It can be disabled)

**Comment by werwolf2303** [comment]
I'm currently working on getting the application fixed. Spotify forbid bearer tokens generated via their internal api to access the web api, so I need to switch completely to the internal api.

**Comment by werwolf2303** [comment]
<img width="1021" height="919" alt="image" src="https://github.com/user-attachments/assets/131947ef-ec96-4e2f-9122-c9c72122cfe5" />
Not all of the features are restored. Only the library is half working right now

**Comment by werwolf2303** [comment]
I got librespot, the rust version to compile on Windows XP. Together with egui it just uses 20MB of RAM. I'm not sure yet if I will do a complete rewrite of NTify, just showing that it's possbile.
<img width="1442" height="1222" alt="Screenshot_20260902_224727" src="https://github.com/user-attachments/assets/492dc599-6100-4905-97fe-ef0f252d0998" />
